### PR TITLE
Add support for generic IP protocols.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

* Allow such packets through to policy.
* Add conntrack support with 10 min timer (based in Linux default).
* Add UTs for policy matches.
* Add FVs that verify protocol match and conntrack works as expected for host endpoints.
  * Extend `test-connection` and `test-workload` to support arbitrary protocols
  * Clean up `test-workload`'s protocol flags
  * Ignore ports for raw IP protocol tests

## Todos
- [x] Unit tests (full coverage)
- [x] FV tests
- [x] Release note

## Release Note

```release-note
In eBPF mode, generic IP protocols (i.e. everything apart from TCP, UDP and ICMP) are now supported in a limited fashion (similarly to iptables). Such protocols are connection-tracked based on IP and protocol number only.
```

### CODE REVIEW

File: fv/test-workload/test-workload.go [hunk 1]
```diff
@@ -394,60 +393,34 @@ func main() {
...
+				go loopRespondingToPackets(logCxt, p)
+			} else if protocol == "udp" {
 				// Since UDP is connectionless, we can't use Listen() as we do for TCP.  Instead,
 				// we use ListenPacket so that we can directly send/receive individual packets.
 				logCxt.Info("About to listen for UDP packets")
 				p, err := net.ListenPacket("udp", myAddr)
 				panicIfError(err)
 				logCxt.Info("Listening for UDP connections")
 
-				go func() {
```
Comment (fasaxc) [inline fv/test-workload/test-workload.go:414]: This code just moved down below as a named function (so it can be used for UDP and raw IP)

**Review by tomastigera — COMMENTED** [review]
LGTM let me know when the broken test is fixed

File: fv/hostendpoints_test.go [hunk 2]
```diff
@@ -288,6 +312,60 @@ func describeHostEndpointTests(getInfra infrastructure.InfraFactory, allInterfac
...
+				Protocol: &proto253,
+			}}
+			policy.Spec.Selector = fmt.Sprintf("hostname == '%s'", felixes[1].Hostname)
+			_, err = client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+			Expect(err).NotTo(HaveOccurred())
+
+			// One-way check because conntrack for unknown protocols is symmetric so, once the
+			// felix[0] -> felix[1] packet goes through, we expect felix[1] -> felix[0] to work too
+			// (and the connetivity checker always does a request-response to check that).
+			cc253.Expect(connectivity.None, felixes[0], rawIPHostW253[1])
```
[the remark this answers is not in the document]
Comment (fasaxc) [reply fv/hostendpoints_test.go:364]: Eek

**Comment by fasaxc** [comment]
@tomastigera all good now; breakage was caused by being more strict in the connectivity checker in interpreting the output of `test-connection`.  Turned out we had a couple of special-case tests that try to connect to the API server and typha using a raw TCP socket; of course, they don't expect any sensible response and that lack of response now made the test fail.

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by tomastigera — APPROVED** [review]

**Pull request merged**